iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,200 +37,299 @@ class Duration;
 
-/// @brief constructs a new Duration object in microseconds
-constexpr Duration operator"" _us(unsigned long long int); // PRQA S 48
+/// @brief Constructs a new Duration object from seconds
+constexpr Duration operator"" _s(unsigned long long int) noexcept; // PRQA S 48
 
-/// @brief constructs a new Duration object in milliseconds
-constexpr Duration operator"" _ms(long double);
+/// @brief Constructs a new Duration object from minutes
+constexpr Duration operator"" _m(unsigned long long int) noexcept; // PRQA S 48
 
-/// @brief constructs a new Duration object in milliseconds
-constexpr Duration operator"" _ms(unsigned long long int); // PRQA S 48
+/// @brief Constructs a new Duration object from hours
+constexpr Duration operator"" _h(unsigned long long int) noexcept; // PRQA S 48
 
-/// @brief constructs a new Duration object in seconds
-constexpr Duration operator"" _s(long double);
-
-/// @brief constructs a new Duration object in seconds
-constexpr Duration operator"" _s(unsigned long long int); // PRQA S 48
-
-/// @brief constructs a new Duration object in minutes
-constexpr Duration operator"" _m(long double);
-
-/// @brief constructs a new Duration object in minutes
-constexpr Duration operator"" _m(unsigned long long int); // PRQA S 48
-
-/// @brief constructs a new Duration object in hours
-constexpr Duration operator"" _h(long double);
-
-/// @brief constructs a new Duration object in hours
-constexpr Duration operator"" _h(unsigned long long int); // PRQA S 48
-
-/// @brief constructs a new Duration object in days
-constexpr Duration operator"" _d(long double);
-
-/// @brief constructs a new Duration object in days
-constexpr Duration operator"" _d(unsigned long long int); // PRQA S 48
+/// @brief Constructs a new Duration object from days
+constexpr Duration operator"" _d(unsigned long long int) noexcept; // PRQA S 48
 } // namespace duration_literals
 
 /// @code
 ///   #include <iostream>
 ///   // ...
 ///   using namespace units;
-///   auto timeInDays = 12_d;
-///   auto timeInSeconds = 1.5_s;
-///   std::cout << timeInDays << std::endl;
-///   std::cout << timeInDays.nanoSeconds<int>() << " ns" << std::endl;
-///   std::cout << timeInSeconds.minutes<float>() << " min" << std::endl;
+///   using namespace units::duration_literals;
+///   auto someDays = 2 * 7_d + 5_ns;
+///   auto someSeconds = 42_s + 500_ms;
+///   std::cout << someDays << std::endl;
+///   std::cout << someDays.nanoSeconds<uint64_t>() << " ns" << std::endl;
+///   std::cout << someSeconds.milliSeconds<int64_t>() << " ms" << std::endl;
 /// @endcode
 class Duration
 {
   public:
+    // BEGIN CREATION FROM STATIC FUNCTIONS
+
+    /// @brief Constructs a new Duration object from nanoseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as nanoseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
```

Comment:
How about "clamped to"?

Reply:
will be fixed to 0?

Reply:
other possibilities would be
- bounded to 0
- limited to 0
- narrowed to 0

Maybe we need the help of a native speaker @ithier

Reply:
I do not really care too much about the wording, it just caught my attention while reading. I think clamp is popular in computer graphics but maybe otherwise unheard of. Do whatever.